src/guidescanpy/flask/core/genome.py

```diff
@@ -342,6 +344,29 @@ def query(
             if max_gc is not None:
                 results = results[results["gc-content"] <= max_gc]
 
+            if pattern_avoid is not None:
+                wildcard_to_nuc = {"N": "ACTG", "V": "ACG"}
+                patterns_avoid = [pattern_avoid]
+
+                for wildcard in wildcard_to_nuc:
+                    patterns_avoid.extend(
+                        [
+                            pattern.replace(wildcard, x)
+                            for x in wildcard_to_nuc[wildcard]
+                            for pattern in patterns_avoid
+                            if wildcard in pattern
+                        ]
+                    )
+
+                patterns_avoid = [
+                    pattern
+                    for pattern in patterns_avoid
```

Comment:
This is the part I was trying to avoid with my suggestion with the refactoring - you're going through the list to find out and remove the ones with the wildcard, even though we know that if it's present it's going to be the very first one, so you might as well handle that case in an if/else.

Reply:
The statement "if it's present it's going to be the very first one" is not necessarily true, because in some very rare case `N` and `V` can both appear in one pattern. For example, if the input pattern is `AVNV`, the `patterns_avoid` before line 361 would be 
```['AVNV', 'AVAV', 'AVCV', 'AVTV', 'AVGV', 'AANA', 'AAAA', 'AACA', 'AATA', 'AAGA', 'ACNC', 'ACAC', 'ACCC', 'ACTC', 'ACGC', 'AGNG', 'AGAG', 'AGCG', 'AGTG', 'AGGG']```.

I modified the logic so we can get rid of the removing part.